canard/subscriber.h

```diff
@@ -41,62 +41,34 @@ class Subscriber : public HandlerList {
     Subscriber(Callback<msgtype> &_cb, uint8_t _index) NOINLINE_FUNC :
     HandlerList(CanardTransferTypeBroadcast, msgtype::cxx_iface::ID, msgtype::cxx_iface::SIGNATURE, _index),
     cb (_cb) {
-#ifdef WITH_SEMAPHORE
-        WITH_SEMAPHORE(sem[index]);
-#endif
-        next = branch_head[index];
-        branch_head[index] = this;
-        link(); // link ourselves into the handler list now that we're in the branch list
+        // link ourselves into the handler list
+        link();
     }
 
     // delete copy constructor and assignment operator
     Subscriber(const Subscriber&) = delete;
 
     // destructor, remove the entry from the singly-linked list
     ~Subscriber() NOINLINE_FUNC {
-#ifdef WITH_SEMAPHORE
-        WITH_SEMAPHORE(sem[index]);
-#endif
-        unlink(); // unlink ourselves from the handler list before the branch list
-        Subscriber<msgtype>* entry = branch_head[index];
-        if (entry == this) {
-            branch_head[index] = next;
-            return;
-        }
-        while (entry != nullptr) {
-            if (entry->next == this) {
-                entry->next = next;
-                return;
-            }
-            entry = entry->next;
-        }
+        // unlink ourselves from the handler list
+        unlink();
     }
 
     /// @brief parse the message and call the callback
     /// @param transfer transfer object
-    void handle_message(const CanardRxTransfer& transfer) override NOINLINE_FUNC {
+    bool handle_message(const CanardRxTransfer& transfer) override NOINLINE_FUNC {
```

Comment:
Should return `true` iff decode succeeds.

Reply:
not quite, one of the advantages of lua DroneCAN is we can fix incorrect DSDL. We have had that from vendors, where DSDL that we don't control (eg. Hobbywing) changes. If we try to short circuit on decode failure then a lua receiver can't fix the issue

Reply:
Okay fair enough, then it should always return `false`.